app/main/views/index.py

```diff
@@ -53,16 +58,49 @@ def landing(service_id, document_id):
     )
 
 
+@main.route('/d/<base64_uuid:service_id>/<base64_uuid:document_id>/confirm-email-address', methods=['GET', 'POST'])
+def confirm_email_address(service_id, document_id):
+    key = request.args.get('key')
+    if not key:
+        abort(404)
+
+    service = _get_service_or_raise_error(service_id)
+
+    service_contact_info = service['data']['contact_link']
+    contact_info_type = assess_contact_type(service_contact_info)
+
+    if not _get_document_metadata(service_id, document_id, key):
+        return render_template(
+            'views/file_unavailable.html',
+            service_name=service['data']['name'],
+            service_contact_info=service_contact_info,
+            contact_info_type=contact_info_type,
+        )
+
+    form = EmailAddressForm()
+
+    if form.validate_on_submit():
+        return redirect(url_for('.download_document', service_id=service_id, document_id=document_id, key=key))
+
+    show_invalid_email_error = ValidEmail.message in form.email_address.errors
```

Comment:
Is this how we normally handle populating the error summary component?

Reply:
I'm the first person to use the error summary component in our code (we have a story to add in in admin) 😀. I have changed how we populate it though, because I realised that we need to show the error summary for all errors and it wasn't showing for the case where someone had left the email address field blank. It's now a bit more "general".

Reply:
Great :)

Reply:
I've got another possible implementation that might be a bit cleaner and more general (and a good template for us to copy and paste into admin later). Let's have a chat over it :)

Reply:
The macro discussed is now added in a new commit